ibis/backends/dask/execution/util.py

```diff
@@ -47,25 +48,103 @@ def make_selected_obj(gs: SeriesGroupBy):
         ]
 
 
-def maybe_wrap_scalar(result: Any, expr: ir.Expr) -> Any:
-    """
-    A partial implementation of `coerce_to_output` in the pandas backend.
+def coerce_to_output(
+    result: Any, expr: ir.Expr, index: Optional[pd.Index] = None
+) -> Union[dd.Series, dd.DataFrame]:
+    """Cast the result to either a Series of DataFrame, renaming as needed.
+
+    Reimplementation of `coerce_to_output` in the pandas backend, but
+    creates dask objects and adds special handling for dd.Scalars.
+
+    Parameters
+    ----------
+    result: Any
+        The result to cast
+    expr: ibis.expr.types.Expr
+        The expression associated with the result
+    index: pd.Index
+        Optional. If passed, scalar results will be broadcasted according
+        to the index.
 
-    Currently only wraps scalars, but will change when udfs are added to the
-    dask backend.
+    Returns
+    -------
+    result: A Series or DataFrame
     """
     result_name = expr.get_name()
-    if isinstance(result, dd.core.Scalar) and isinstance(
-        expr.op(), ops.Reduction
-    ):
-        # TODO - computation
-        return dd.from_pandas(
-            pd.Series(result.compute(), name=result_name), npartitions=1
-        )
+    if isinstance(expr, (ir.DestructColumn, ir.StructColumn)):
+        result = safe_concat(list(result))
+        result.columns = expr.type().names
+        return result
+    elif isinstance(expr, (ir.DestructScalar, ir.StructScalar)):
+        return _coerce_to_dataframe(result, expr.type().names)
+    elif isinstance(result, dd.Series):
+        result.rename(result_name)
+    if isinstance(expr.op(), ops.Reduction):
+        if isinstance(result, dd.Series):
+            return result.rename(result_name)
+        elif isinstance(result, dd.core.Scalar):
+            # TODO - computation
+            return dd.from_pandas(
+                pd.Series(result.compute(), name=result_name), npartitions=1
+            )
+        else:
+            series = dd.from_array(np.repeat(result, len(index)))
+            series.name = result_name
+            series.index = index
+            return series
+
     else:
         return result.rename(result_name)
 
 
+def _coerce_to_dataframe(data: Any, names: List[str]) -> pd.DataFrame:
```

Comment:
Return type is wrong?

Reply:
Can you add in the docstring what are the valid input types?

Reply:
definitly add Parameters / Returns as much as possible. wouldn't object to Examples if useful.